### PR TITLE
:zap: (synced-prefs) preloading in redux state to improve perf

### PR DESCRIPTION
**Pull request by MatissJanis**
https://github.com/actualbudget/actual/issues/3524

### CODE REVIEW

File: packages/loot-core/src/client/state-types/prefs.d.ts [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
   local: MetadataPrefs;
   global: GlobalPrefs;
+  synced: SyncedPrefs;
 };
 
 export type SetPrefsAction = {
   type: typeof constants.SET_PREFS;
   prefs: MetadataPrefs;
   globalPrefs: GlobalPrefs;
+  syncedPrefs: SyncedPrefs;
```
[the remark this answers is not in the document]
Comment (MatissJanis) [reply packages/loot-core/src/client/state-types/prefs.d.ts:18]: False. The reducer has been updated already.

File: packages/loot-core/src/client/actions/sync.ts [hunk 2]
```diff
@@ -45,8 +43,12 @@ export function sync() {
       if ('error' in result) {
         return { error: result.error };
       }
-      return {};
+
+      // Update the prefs
+      await dispatch(loadPrefs());
     }
+
+    return {};
```
[the remark this answers is not in the document]
Comment (MatissJanis) [reply packages/loot-core/src/client/actions/sync.ts:51]: No thanks

File: packages/loot-core/src/client/actions/sync.ts [hunk 3]
```diff
@@ -45,8 +43,12 @@
       if ('error' in result) {
         return { error: result.error };
       }
-      return {};
+
+      // Update the prefs
+      await dispatch(loadPrefs());
```
[the remark this answers is not in the document]
Comment (MatissJanis) [reply packages/loot-core/src/client/actions/sync.ts:48]: this is fine

File: packages/loot-core/src/server/preferences/app.ts [hunk 4]
```diff
@@ -18,4 +18,17 @@ const savePreferences = async ({
...
+const getPreferences = async (): Promise<SyncedPrefs> => {
+  const prefs = (await db.all('SELECT id, value FROM preferences')) as Array<{
+    id: string;
+    value: string;
+  }>;
+
+  return prefs.reduce<SyncedPrefs>(
+    (carry, { value, id }) => ({ ...carry, [id]: value }),
+    {},
+  );
```
[the remark this answers is not in the document]
Comment (MatissJanis) [reply packages/loot-core/src/server/preferences/app.ts]: Done

File: packages/desktop-client/src/hooks/useSyncedPrefs.ts [hunk 5]
```diff
@@ -1,39 +1,22 @@
...
-import { send } from 'loot-core/platform/client/fetch';
-import { q } from 'loot-core/shared/query';
+import { saveSyncedPrefs } from 'loot-core/client/actions';
+import { type State } from 'loot-core/client/state-types';
 import { type SyncedPrefs } from 'loot-core/src/types/prefs';
 
 type SetSyncedPrefsAction = (value: Partial<SyncedPrefs>) => void;
 
 /** @deprecated: please use `useSyncedPref` (singular) */
 export function useSyncedPrefs(): [SyncedPrefs, SetSyncedPrefsAction] {
```
[the remark this answers is not in the document]
Comment (MatissJanis) [reply packages/desktop-client/src/hooks/useSyncedPrefs.ts:11]: Done

**Comment by youngcw** [comment]
The  privacy mode loading is fixed.  I cant tell yet if this also helped the goals info or if its just that its hard to see properly.

**Comment by MatissJanis** [comment]
Not sure what the problem with goal info would be, but this PR will solve the slow-ness of SyncedPrefs (so privacy mode and such). Open for review now :) 

Looking for 2+ reviewers here.

**Review by MikesGlitch — APPROVED** [review]
seems faster :shipit:

**Comment by youngcw** [comment]
I think this entirely fixes that issue

**Comment by MatissJanis** [comment]
Looking for 1 more review here before merging.

**Review by csenel — APPROVED** [review]

**Review by jfdoming — APPROVED** [review]

**Pull request merged**